Breeze2

```diff
@@ -41,11 +43,21 @@ def save_config():
 
 if needs_installation():
     print(f"(Re)Installing Breeze's virtualenv in {BUILD_BREEZE_VENV_DIR}")
-    BUILD_BREEZE_VENV_DIR.mkdir(parents=True, exist_ok=True)
-    subprocess.run([sys.executable, "-m", "venv", f"{BUILD_BREEZE_VENV_DIR}"], check=True)
-    subprocess.run(
-        [f"{BUILD_BREEZE_VENV_PIP}", "install", "--upgrade", "-e", "."], cwd=BREEZE_SOURCE_PATH, check=True
+    EnvBuilder(system_site_packages=False, upgrade=True, with_pip=True, prompt="breeze").create(
+        str(BUILD_BREEZE_VENV_DIR)
     )
+    if os.name == 'nt':
+        subprocess.run(
+            [f"{BUILD_BREEZE_VENV_PYTHON}.exe", "-m", "pip", "install", "--upgrade", "-e", "."],
+            cwd=BREEZE_SOURCE_PATH,
+            check=True,
+        )
+    else:
+        subprocess.run(
+            [f"{BUILD_BREEZE_VENV_PYTHON}", "-m", "pip", "install", "--upgrade", "-e", "."],
+            cwd=BREEZE_SOURCE_PATH,
+            check=True,
+        )
```

Comment:
Windows appends the executable extension automatically, so you don’t need to append `.exe` on your own. The behaviour is not exactly the same—it uses `PATHEXT`, which contains more than just `.exe`, so theoratically this may pick up a `python.bat`—but only with negligible difference.

Reply:
Cool.

Reply:
Actually I want to add back `python -m venv` as fallback. I noticed that I had problems with making the API works on MacOS when I had default MacOS Python set as "global" Python rather than the one coming from Pyenv. Configuring Pyevn to work properly on MacOS is kinda not obvious in this case. 

It got fixed when I set `pyenv global`

The problem I had was that "ensurepip" was missing. 

Is it a known  issue @uranusjr ? I might want to reproduce it and possibly report it.

Reply:
BTW. IS there an API (PIP PI I guess) I could also use to install the needed packages into the new venv? Kinda strange to use subprocess for that.

Reply:
OK. Got Windows installed in VirtualBox now so I will do seme more tests with it.

Reply:
Not with venv (it is very barebone), but if you use virtualenv there is a thing called [seeders](https://virtualenv.pypa.io/en/latest/cli_interface.html#section-seeder).

Installing things on environment creation is out of pip’s control because it needs hooks in the environment builder, and venv does not provide that.

Reply:
Some more info:  the main problem was "trying" to create venv from within the "frozen" python script (in bundled .exe file). I debugged it quite a bit  and I've learned quite a lot about those things. 

The "frozen" python binary does not run using executable interpreter. This was a bit surprising -  it actually uses a python interpereter library, which you cannot "execute" modules with. And `sys.executable` is surprisingly the actual .exe file that was run :scream:  (so in our case ./Breeze2.exe).  @edithturn  - that's why you got those repeating "creating environement| messages - the python code simply continued to re-run the `Breeze2.exe` on failure of trying to create an evnironment using the API.

And creating using the API failed for a different reason. First of all ensurepip is run "implicitly" by "venv create" and pip installer when prepares a bundle, only uses modules thata are actually used by the python script (and ensurepip was not detected). 

But even if I used `---hidden-import` flag of the `pyinstaller` and added it (Pyinstaller is actually quite configurable), the `ensurepip` when installing PIP in the environment, also uses the python interpreter executable :scream: and of course gets into another endless loop.